### ISSUE TITLE
Update concourse/dex dependancy to at least v2.39.0 from upstream

### ISSUE THREAD

**Issue by taylorsilva** [concourse/concourse#8970]
### Discussed in https://github.com/concourse/concourse/discussions/8961

<div type='discussions-op-text'>

<sup>Originally posted by **Kump3r** May 22, 2024</sup>
Hello all,
With the following [PR](https://github.com/dexidp/dex/pull/2454), we contributed to the [dexidp/dex](https://github.com/dexidp/dex.git), in order to log failed authentication requests, which is a security requirement to have a mechanism of coping with brute force attacks. I am opening this discussion and I have also written a comment in #5525 and in [discord thread](https://discord.com/channels/219899946617274369/413771057686773770/1242064682345435146). Following [this documentation](https://github.com/concourse/concourse/blob/master/CONTRIBUTING.md#rebuilding-to-test-your-changes), I was able to validate successfully, that the upstream change, will log such attempts in the web logs in the future. Can you provide an estimate as to when a release, containing dex upstream changes from [v2.39.0](https://github.com/dexidp/dex/tree/v2.39.0), would be available for Concourse? Thanks in advance!</div>

---

Someone needs to update our fork here: https://github.com/concourse/dex

**Comment by Kump3r** [comment]
I can try to work on this. I will tag in the PR with details.

**Comment by Kump3r** [comment]
Since I assume maintainers can trigger the automation, I won't be able todo it myself, but the issue is still very relevant and the fact that Concourses dex server can be brute-forced, without any trace seems like a security vulnerability. What do you think?

**Comment by taylorsilva** [comment]
@Kump3r Hey, help would be appreciated! I merged in your PR in the governance repo.

**Comment by Kump3r** [comment]
Hey @taylorsilva thanks! Will that provide access to the pipeline responsible for adopting dex upstream changes? Do I need to request something else as well?

**Comment by taylorsilva** [comment]
No that should be it. We do one issue though in that the infra (provided by Broadcom) is down right now due to a certificate issue between the Concourse and Vault instance. I'm on the outside like you, but seeing who I can poke to get things going on the Broadcom side.

**Comment by Kump3r** [comment]
Okay, thanks for the feedback so far I am able to only see public pipelines, without the fore-mentioned one. But I am still not a member of any team, so I guess that is fine for now. Once I am able to see the pipeline I will check it out and hopefully create a dex fork release soon after. :)

**Comment by Kump3r** [comment]
Do I need to be part of a specific team in order to gain access to the pipeline? I assume since I am able to login with github, at least my permissions would have been populated by now or? Thanks for the help!

**Comment by taylorsilva** [comment]
Yeah I think you do. I need to dive into the governance repo and see what's happening there. Lots to do.

**Comment by Kump3r** [comment]
Something that might be a bit more pressing (or rather depressing) for the dex fork is, that based on [this documentation](https://v2-apidocs.cloudfoundry.org/#users), we will not be able to use the CF authentication connector, unless we migrate to the CF v3 API. Issue spotted [here](https://github.com/concourse/dex/blob/master/connector/cloudfoundry/cloudfoundry.go#L352-L355) and possibly [here](https://github.com/concourse/dex/blob/8337b8fa07291fb9efe61010e51220ece78a9694/connector/cloudfoundry/cloudfoundry.go#L97)

**Comment by Kump3r** [comment]
I created #9006 as to have separation, between both topics.

**Comment by Kump3r** [comment]
As Broadcom infras are back up (to the best of my knowledge), can we revisit the topic, as I still lack access to the [update-dex](https://ci.concourse-ci.org/teams/main/pipelines/update-dex) pipeline in the main team. I was hoping to adopt upstream changes and switch to at least version [v2.39.0](https://github.com/dexidp/dex/tree/v2.39.0), as to kill 2 birds with one stone as they say. Is there a validation process with a new dex service from the upstream we can/should follow?

**Comment by taylorsilva** [comment]
The `update-dex` pipeline wasn't working. The PR's couldn't be automatically rebased onto upstream dex due to merge conflicts. I manually fixed the conflicts and the pipeline works now. Our fork is now rebased on top of dex v2.41.1

Will look into how we update dex in Concourse. I think we need to cut our own version of dex and then update the go dependency.

**Comment by taylorsilva** [comment]
oof, our stuff is spread over too many issues. Closing this issue in favour of #9006 and that our fork is on v2.41.1 now
